components/local-app/BUILD.js

```diff
@@ -21,10 +21,8 @@ const generatePackage = function (goos, goarch, binaryName, mainFile) {
             buildCommand: [
                 "sh",
                 "-c",
-                'go build -trimpath -ldflags "-X github.com/gitpod-io/local-app/pkg/constants.GitCommit=${__git_commit} -X github.com/gitpod-io/local-app/pkg/constants.BuildTime=$(date +%s)" -o ' +
-                    binaryName +
-                    " " +
-                    mainFile,
+                // We need to set GOARCH explicitly here because the `defaultVariant` in `WORKSPACE.yaml` overrides it for the workspace
+                `GOARCH=${goarch} go build -trimpath -ldflags "-X github.com/gitpod-io/local-app/pkg/constants.GitCommit=\${__git_commit} -X github.com/gitpod-io/local-app/pkg/constants.BuildTime=\$(date +%s)" -o ${binaryName} ${mainFile}`,
```

Comment:
isn't this the same with `env` line above?
```
        env: ["GOOS=" + goos, "GOARCH=" + goarch, "CGO_ENABLED=0"],
```

Reply:
Not sure if I follow exactly, but inlining the env variable here means it's out of reach for leeway to change, so we don't have to worry about it being [overridden](https://github.com/gitpod-io/gitpod/blob/2a8ee81ca3ffbcd3809e7ab967ef3b90a40b8f3d/WORKSPACE.yaml#L38). The other vars can be kept because we don't override them anywhere.

Reply:
I see, you mean the bug is caused by `env` defined in current leeway build will be overridden by what's defined in WORKSPACE.yaml

It's a weird behavior, maybe worth create an issue on leeway as well? not sure if it's designed by that

Reply:
Raised https://github.com/gitpod-io/leeway/issues/186 cc @mustard-mh